### PR TITLE
[NON-JIRA] Refactor identity

### PR DESCRIPTION
**Pull request by kacpersmal**
This PR contains new identity platform approach.

### CODE REVIEW

File: src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/AccountController.Password.cs [hunk 1]
```diff
@@ -150,15 +149,13 @@ public IActionResult ChangePassword()
...
     public async Task<IActionResult> ChangePasswordFinish(ChangePasswordViewModel viewModel)
     {
-        if (!ModelState.IsValid)
-        {
-            return View(viewModel);
-        }
+        // if (!ModelState.IsValid)
+        // {
+        //     return View(viewModel);
+        // }
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/AccountController.Password.cs:155]: Is it commented out on purpose, or you forgot to uncomment it?

File: src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/Code/PasswordChangeWithCodeResponse.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace DfT.ZEV.Common.MVC.Authentication.Identity.IdentityApi.Endpoints.PasswordChange.Code;
+
+/// <summary>
+/// Represents a response from a password change request using a code.
+/// </summary>
+public class PasswordChangeWithCodeResponse
+{
+}
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/Code/PasswordChangeWithCodeResponse.cs:8]: Why is it empty?

File: src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/GetCode/GetPasswordResetCodeRequest.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+/// <summary>
+/// Represents a request to get a password reset code.
+/// </summary>
+public class GetPasswordResetCodeRequest
+{
+    /// <summary>
+    /// Gets the request type, which is always "PASSWORD_RESET" for this request.
+    /// </summary>
+    public string RequestType { get; } = "PASSWORD_RESET";
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/GetCode/GetPasswordResetCodeRequest.cs:11]: If it's always the same value, why not `const string`? Won't it work with http client?

File: src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/GetCode/GetPasswordResetCodeRequest.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    /// <summary>
+    /// Gets a value indicating whether to return the out-of-band (OOB) link. This is always true for this request.
+    /// </summary>
+    public bool ReturnOobLink { get; } = true;
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/PasswordChange/GetCode/GetPasswordResetCodeRequest.cs:16]: `const bool`? Ok, but I see that property might just look better here, so it's up to you.

File: src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/Authorization/AuthorizationRequest.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    /// <summary>
+    /// Gets or sets the password used for authorization.
+    /// </summary>
+    public string Password { get; set; } = null!;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether a secure token should be returned.
+    /// </summary>
+    public bool ReturnSecureToken { get; set; } = true;
```
Comment (dae-ne) [inline src/common/DfT.ZEV.Common.MVC.Authentication/Identity/IdentityApi/Endpoints/Authorization/AuthorizationRequest.cs:22]: .

**Review by dae-ne — COMMENTED** [review]
Please, scan the code with stylecop

File: src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/AccountController.Password.cs (hunk 1 above)
Comment (kacpersmal) [reply src/common/DfT.ZEV.Common.MVC.Authentication/Areas/Authentication/Controllers/AccountController.Password.cs]: Forgot but uncommented it

**Comment by cruddasj** [comment]
@kacpersmal - you might want to change the PR title to not reference the JIRA task. Reason being is it will move the story to dev test on merge... but it isn't actually implementing MFA in this PR

**Comment by kacpersmal** [comment]
@cruddasj It will not be merged untill fully developed MFA, i've created this PR so you and @dae-ne can look into it

**Comment by cruddasj** [comment]
> @cruddasj It will not be merged untill fully developed MFA, i've created this PR so you and @dae-ne can look into it

Fair enough 👍

**Pull request merged**